src/components/Reports/GoalCalculator/Shared/calculateTotals.ts

```diff
@@ -0,0 +1,121 @@
+import { PrimaryBudgetCategoryEnum } from 'src/graphql/types.generated';
+import { BudgetFamilyFragment } from './GoalCalculation.generated';
+import type { ListGoalCalculationFragment } from '../GoalsList/GoalCalculations.generated';
+
+const toPercentage = (value: number | null | undefined) => (value ?? 0) / 100;
+
+export interface GoalTotals {
+  netMonthlySalary: number;
+  taxesPercentage: number;
+  taxes: number;
+  salaryPreIra: number;
+  rothContributionPercentage: number;
+  traditionalContributionPercentage: number;
+  rothContribution: number;
+  traditionalContribution: number;
+  grossAnnualSalary: number;
+  grossMonthlySalary: number;
+  benefitsCharge: number;
+  ministryExpensesTotal: number;
+  overallSubtotal: number;
+  overallSubtotalWithAdmin: number;
+  overallTotal: number;
+}
+
+export const calculateGoalTotals = (
+  goalCalculation: ListGoalCalculationFragment | null,
+): GoalTotals => {
+  const netPaycheckAmount = goalCalculation?.netPaycheckAmount ?? 0;
+  const spouseNetPaycheckAmount = goalCalculation?.spouseNetPaycheckAmount ?? 0;
+  const totalNetPaycheckAmount = netPaycheckAmount + spouseNetPaycheckAmount;
+  const taxesPercentage = toPercentage(goalCalculation?.taxesPercentage);
+  const spouseTaxesPercentage = toPercentage(
+    goalCalculation?.spouseTaxesPercentage,
+  );
+  const paySplitPercentage = netPaycheckAmount / totalNetPaycheckAmount || 1;
+  const spousePaySplitPercentage = 1 - paySplitPercentage;
+  const totalTaxesPercentage =
+    paySplitPercentage * taxesPercentage +
+    spousePaySplitPercentage * spouseTaxesPercentage;
+
+  const specialIncomeCategory =
+    goalCalculation?.specialFamily.primaryBudgetCategories.find(
+      (category) =>
+        category.category === PrimaryBudgetCategoryEnum.SpecialIncome,
+    );
+  const additionalIncome = specialIncomeCategory
+    ? calculateCategoryTotal(specialIncomeCategory)
+    : 0;
+  const netMonthlySalary =
+    (goalCalculation
+      ? calculateFamilyTotal(goalCalculation.householdFamily)
+      : 0) - additionalIncome;
```

Comment:
Does this calculation take into account that we get paid twice a month?
So your $2500 + spouse $2000 would be $11000 rather than $5500

Reply:
The calculations from here on down all deal with monthly salaries not paycheck amounts. The `netPaycheckAmount` calculations at the top are just to calculate `paySplitPercentage`.

Reply:
The calculations from here on down all deal with monthly salaries not paycheck amounts. The `netPaycheckAmount` calculations at the top are just to calculate `paySplitPercentage`.